db/db_iter.cc

```diff
@@ -390,6 +400,12 @@ void DBIter::FindNextUserEntryInternal(bool skipping, bool prefix_check) {
       break;
     }
 
+    if (TooManyInternalKeysSkipped()) {
```

Comment:
let's look into if it's possible to have it look like this
```
if (BumpInternalKeysSkipped()) {
  return false;
}
```
we can even have a statistics counter inside this function that will report the total number of internal keys that we need to skip